docs/test-faster.md

```diff
@@ -0,0 +1,191 @@
+# Testing faster: How to avoid making compile times worse by adding tests
```

Comment:
I don't feel strongly about it, but I wonder if this doc might suit living as a "design doc" in the `./design/` subdirectory here.